duffel_api/http_client.py

```diff
@@ -61,7 +61,7 @@ def __iter__(self):
         while "meta" in response:
             after = response["meta"]["after"]
             for entry in response["data"]:
-                yield self._caller(entry)
+                yield self._caller.from_json(entry)
```

Comment:
@richcooper95: I'm curious how we could ensure that the _thing_ here being called has the `from_json`.

Reply:
If `self._caller` is the class we want to instantiate, and we _only_ want to allow `self._caller` to be set if it contains a `from_json` classmethod, then we could add validation* to the `self._caller` attribute - something along the lines of:

```python
from inspect import getattr_static

try:
    assert isinstance(getattr_static(x, "from_json"), classmethod)
except (AttributeError, AssertionError) as _:
    raise TypeError("'caller' must have a 'from_json' classmethod") from None
```

*might be a good opportunity to try [this method of validation](https://www.notion.so/duffel/Python-SDK-db0afb7cafe74c05ba5c2dc9fcbc722e#03512c224cf943829bd6a86a4e642f37)?

Reply:
Did you have any thoughts on this?

Reply:
I tried to reply by email, but it hasn't shown up!

Basically, I'm going to leave this for another PR, and have a proper think about it.